src/main/java/org/kitteh/irc/client/library/implementation/NettyManager.java

```diff
@@ -286,18 +305,108 @@ public String toString() {
         }
     }
 
+    static class DCCConnection extends ChannelInitializer<SocketChannel> {
+
+        private final IRCDCCExchange exchange;
+        private final InternalClient client;
+        // Only allow one connection per DCC. Weird, I know.
+        private boolean oneConnection;
+
+        private DCCConnection(IRCDCCExchange ex, InternalClient client) {
+            this.exchange = ex;
+            this.client = client;
+        }
+
+        @Override
+        public void initChannel(SocketChannel channel) throws Exception {
+            if (this.oneConnection) {
+                channel.close();
+                return;
+            }
+            this.oneConnection = true;
+            this.exchange.setNettyChannel(channel);
+            dccConnections.computeIfAbsent(this.client, c -> new ArrayList<>()).add(channel);
```

Comment:
It's actually NettyManager.dccConnections. Do you still want fully qualified for outer static fields?